### PR TITLE
UOS Qr Support (rebased on master)

### PR DESCRIPTION
**Pull request by yjkimjunior**
I made a new branch for rebasing #273 rather than force pushing. Unit tests pass and the tested the same QRs from https://uostestqrs.herokuapp.com/ as well on iOS.

* [x] - QrView: would not change. However in its usage for displaying addresses, it is expected to be in `UOS introduction` format, with extra information at the end (either a `chainId` or a `genesisHash`)
* [x] - QrScanner: 
  * [x] - convert rawBytes from RNNative to U8a **(Unit Tested)**
  * [x] - strip padding, length, frame info **(Unit Tested)**
  * [x] - parse address, crypto (if substrate), payload, data as per UOS specification *(Unit Tested)*
  * [x] - encode address from bytes to ss58 **(Unit Tested)**
  * [x] - merging in from the identicon PR thank you @Tbaut for this one!
  * [x] - decode SCALE payload
    * [x] - Cold Signer SHOULD decode the transaction details from the SCALE encoding and display them to the user for verification before signing. **(Unit Tested)**
      * payload MUST be the SCALE encoding of the tuple of transaction items (nonce, call, era_description, era_header).
      * payload_hash MUST be the Blake2s 32-byte hash of the SCALE encoding of the tuple of transaction items (nonce, call, era_description, era_header).
      * immortal_payload MUST be the SCALE encoding of the tuple of transaction items (nonce, call).
    * [x] - Cold Signer SHOULD attempt to decode the message as UTF-8 encoded human readable string by whatever heuristics it finds suitable *(Unit Tested)* and display it to the user for verification before signing.
    * [x] - Cold Signer SHOULD warn the user that signing a hash is inherently insecure, in the cash of type 01. 
    * [x] - Cold Signer SHOULD (at the user's discretion) sign the message, immortal_payload, or payload if payload is of length 256 bytes or fewer. 
    * [x] - If payload is longer than 256 bytes, then it SHOULD instead sign the Blake2s hash of payload.
    * [x] - Cold Signer SHOULD display all account id values in SS58Check encoding. [#293 - It should be stored in AccountStore as such]
* [x] - ScannerStore: 
  * [x] - setDataToSign with appropriate native method based on network type, i.e. was crypto field defined (substrate) or not (eth)  **(Unit Tested)**
  * [x] - setTxToSign with appropriate hashing method based on network type. `keccak` for Ethereum, `sr25519` ~or `ed25519` for Substrate.~

### CODE REVIEW

File: src/components/PayloadDetailsCard.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+}
+
+function ExtrinsicPart({ label, style, value }) {
+
+  return (
+    <View style={[{ justifyContent: 'center', alignItems: 'flex-start' }, style]}>
+      <View
+        style={{ padding: 5, paddingVertical: 2 }}
+      >
+        <Text style={{ backgroundColor: colors.bg, color: colors.card_bg, textAlign: 'left', fontSize: 20, fontWeight: '800' }}>
```
Comment (hanwencheng) [inline src/components/PayloadDetailsCard.js]: Please use font standard from `font.js`, and include also `font-family`.

File: src/components/PayloadDetailsCard.js [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+              <ExtrinsicPart label='Nonce' value={payload.nonce.toString()} />
+              <ExtrinsicPart label='Tip' value={payload.tip.toString()} />
+              <ExtrinsicPart label='Genesis Hash' value={payload.genesisHash.toString()} />
+            </View>
+          )
+        }
+        {
+          signature && (
+            <View style={{ padding: 5, paddingVertical: 2 }}>
+              <Text style={{ backgroundColor: colors.bg, color: colors.card_bg, textAlign: 'left', fontSize: 20, fontWeight: '800' }}>
```
Comment (hanwencheng) [inline src/components/PayloadDetailsCard.js]: And here also please use `fonts.js`

File: src/components/PayloadDetailsCard.js [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+                Signature
+              </Text>
+              <Text style={{ color: colors.card_bg_text }}>{signature}</Text>
```
Comment (hanwencheng) [inline src/components/PayloadDetailsCard.js]: `fonts` 😼

File: src/screens/SignedTx.js [hunk 4]
```diff
@@ -49,34 +52,51 @@ export default class SignedTx extends React.PureComponent {
...
-          networkKey={this.props.recipient.networkKey || ''}
-          title={this.props.recipient.name}
-        />
+        {
+          sender.protocol === NetworkProtocols.ETHEREUM
+            ? (
+              <React.Fragment>
+                <TxDetailsCard
+                  style={{ marginBottom: 20 }}
+                  description="After scanning and publishing you will have sent"
```
Comment (hanwencheng) [inline src/screens/SignedTx.js]: make the duplicated text into constant maybe?

File: src/stores/ScannerStore.js [hunk 5]
```diff
@@ -15,57 +15,114 @@
...
+    }
+
+    this.setState({
+      unsignedData: parsedData
+    });
+  }
+
+  async setPartData(frame, frameCount, partData, accountsStore) {
+    if (partData[0] === new Uint8Array([0x00]) || partData[0] === new Uint8Array([0x7B])) {
+      // part_data for frame 0 MUST NOT begin with byte 00 or byte 7B.
```
Comment (hanwencheng) [inline src/stores/ScannerStore.js]: why not zero, does zero means [Multipart Payload](https://github.com/maciejhirsz/uos#multipart-payload)

File: src/stores/ScannerStore.js [hunk 6]
```diff
@@ -15,57 +15,114 @@
...
+      
+      nextDataState[frame] = partData;
+
+      this.setState({
+        multipartData: nextDataState
+      });
+    }
+
+    // all the frames are filled
+    if (Object.keys(this.state.multipartData.length) === frameCount) {
```
Comment (hanwencheng) [inline src/stores/ScannerStore.js]: before we have the code:
```
 nextDataState[frame] = partData;
```
so the `this.state.multipartData` is filled not always from the first element, for example, check out the following example:
Screenshot 2019-08-22 at 15 22 54

File: src/stores/ScannerStore.js (hunk 5 above)
Comment (yjkimjunior) [reply src/stores/ScannerStore.js]: Yes, if 0th byte for part data is 0, then the part data will itself be treated as multipart and split it again, which is not the intended behavior. https://github.com/maciejhirsz/uos#multipart-payload

File: src/stores/ScannerStore.js (hunk 6 above)
Comment (yjkimjunior) [reply src/stores/ScannerStore.js]: Good point, I'll fix that with #322. For now I'll comment out the multipart part for this PR while I work on getting some test QRs ready.

File: src/screens/MessageDetails.js [hunk 7]
```diff
@@ -104,11 +106,28 @@ export class MessageDetailsView extends React.PureComponent {
...
+        </Text>        
         <Button
           buttonStyles={{ height: 60 }}
           title="Sign Message"
-          onPress={() => onNext()}
+          onPress={() => {
+            isOversized
+              ? Alert.alert(
+                  "Warning",
+                  "You are signing a transaction hash, which is inherently unsafe. If possible, it is recommended that you construct a multipart payload instead.",
```
[the remark this answers is not in the document]
Comment (Tbaut) [reply src/screens/MessageDetails.js]: I initially thought that this comment got lost. whatever we chose, it should be the same message for message and transactions.

**Comment by Tbaut** [comment]
When linking or unlinking rn-camera I get an error:
```bash
react-native link react-native-camera
error Linking "react-native-camera" failed. Run CLI with --verbose flag for more details.
SyntaxError: Expected [\n\r] but end of input found.
    at peg$buildStructuredError (/home/thib/github/parity-signer/node_modules/xcode/lib/parser/pbxproj.js:429:12)
    at Object.peg$parse [as parse] (/home/thib/github/parity-signer/node_modules/xcode/lib/parser/pbxproj.js:1891:11)
    at pbxProject.parseSync (/home/thib/github/parity-signer/node_modules/xcode/lib/pbxProject.js:63:24)
    at isInstalled (/home/thib/github/parity-signer/node_modules/@react-native-community/cli-platform-ios/build/link/isInstalled.js:44:67)
    at Object.isInstalled (/home/thib/github/parity-signer/node_modules/@react-native-community/cli-platform-ios/build/link/common/isInstalled.js:23:35)
    at Object.keys.forEach.platform (/home/thib/github/parity-signer/node_modules/@react-native-community/cli/build/commands/link/linkDependency.js:53:36)
    at Array.forEach (<anonymous>)
    at linkDependency (/home/thib/github/parity-signer/node_modules/@react-native-community/cli/build/commands/link/linkDependency.js:36:32)
    at process._tickCallback (internal/process/next_tick.js:68:7)
    at Function.Module.runMain (internal/modules/cjs/loader.js:834:11)
```

I can manually add the linking for android sure, but since this error is in xocde.. better let you know

File: android/app/src/main/java/io/parity/signer/MainApplication.java [hunk 8]
```diff
@@ -3,17 +3,16 @@
 import android.app.Application;
 
 import com.facebook.react.ReactApplication;
+import com.reactlibrary.RNSecureStoragePackage;
+import com.swmansion.gesturehandler.react.RNGestureHandlerPackage;
 import com.horcrux.svg.SvgPackage;
 import com.bitgo.randombytes.RandomBytesPackage;
 import com.reactnativecommunity.netinfo.NetInfoPackage;
-import org.reactnative.camera.RNCameraPackage;
```
[the remark this answers is not in the document]
Comment (Tbaut) [reply android/app/src/main/java/io/parity/signer/MainApplication.java]: I pushed an update

File: src/screens/About.js [hunk 9]
```diff
@@ -31,7 +31,7 @@ export default class About extends React.PureComponent {
   render() {
     return (
       <ScrollView style={styles.body} contentContainerStyle={{ padding: 20 }}>
-        <Text style={styles.title}>PARITY SIGNER ({packageJson.version})</Text>
+        <Text style={styles.title}>PARITY SIGNER</Text>
```
[the remark this answers is not in the document]
Comment (yjkimjunior) [reply src/screens/About.js]: that must have been a rebase issue, thanks

**Comment by yjkimjunior** [comment]
| ### Sign Message     | ### Sign Hash         | ### Sign Extrinsic Payload  |
| ------------- |:-------------:| -----:|
| ezgif-2-4421c6d52e6c | ezgif-2-96f04dc4aa26 | ezgif-1-1ebae92564c0 |

File: src/stores/ScannerStore.js [hunk 10]
```diff
@@ -15,57 +15,114 @@
...
+      // fixme: this needs to be concated to a binary blob
+      const concatMultipartData = Object.keys(this.state.multipartData).reduce((result, data) => res.concat(this.state.multipartData[data]));
+      const data = this.setParsedData(concatMultipartData);
+      this.setData(data);
```
Comment (Tbaut) [inline src/stores/ScannerStore.js]: This looks weird to me, setData takes an accountsStore as argument.

**Pull request merged**